ts/tat-trend.ts

```diff
@@ -403,6 +417,8 @@ function getSelectedDataType(): string {
 }
 
 document.addEventListener("DOMContentLoaded", () => {
+  const plotContainer = document.getElementById("plotContainer");
+  if (!plotContainer) return;
```

Comment:
What is the purpose of this check? Should it be an error?

Reply:
if not it throws an uncaught error on the QC report page `Uncaught Error: Required element weekButton not found` for all the buttons and dataSelection elements. It doesn't interfere with any rendering but I just wanted to get rid of the error.

Reply:
So that means `DOMContentLoaded` is firing multiple times, and sometimes it's before the DOM content is loaded..? I'm not sure why that would be, but it seems this is the wrong event to use. Would `load` work better?

Reply:
load did not work, but I can add the checks for elements